contracts/Staking.sol

```diff
@@ -165,6 +156,30 @@ contract Staking is Autopetrified, ERCStaking, ERCStakingHistory, IStakingLockin
         _decreaseLockAmountUnsafe(_from, msg.sender, _amount);
     }
 
+    /**
+     * @notice Transfer `@tokenAmount(stakingToken: address, _allowance)` from `_from`'s lock by `msg.sender` to `_to` (unstaked)
+     * @param _from Owner of locked tokens
+     * @param _to Recipient of the tokens
+     * @param _amount Number of tokens to be transferred
+     */
+    function transferFromLockAndUnstake(
```

Comment:
What do think about renaming this to `unstakeAndTransferFromLock` to keep it consistent with the one we have for `decreaseAndTransferFromLock`?

Reply:
Well,  but the thing is that the unstaking happens after the transfer, as it’s the receiver who unstakes. To me this other name suggests that it’s the sender who unstakes.